staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -185,98 +230,154 @@ func (q *delayingType) AddAfter(item interface{}, duration time.Duration) {
 // expired item sitting for more than 10 seconds.
 const maxWait = 10 * time.Second
 
-// waitingLoop runs until the workqueue is shutdown and keeps a check on the list of items to be added.
-func (q *delayingType) waitingLoop() {
-	defer utilruntime.HandleCrash()
-
-	// Make a placeholder channel to use when there are no items in our list
-	never := make(<-chan time.Time)
+// waitingLoop implements a single goroutine responsible for managaing the arrival of items into the
```

Comment:
The waitingLoop() → waitingLoop#run refactor is really difficult to review / verify has not changed behavior for existing callers. Is there a way to separate the refactor into clearly no-op steps that can be reviewed independent of the new functionality?

Reply:
Probably not easily, we went through a few big revisions to get it to this state.

Reply:
I've removed the waitingLoop refactor which makes the diff quite a lot easier to read.  I'm happy to follow up this PR with the re-factor if you still want it.